src/os/CMakeLists.txt

```diff
@@ -128,3 +129,29 @@ endif()
 if(WITH_LTTNG AND WITH_EVENTTRACE)
   add_dependencies(os eventtrace_tp)
 endif()
+
+if(WITH_LIBURING)
+  include(ExternalProject)
+  if("${CMAKE_GENERATOR}" MATCHES "Make")
+    set(make_cmd "$(MAKE)")
+  else()
+    set(make_cmd "make")
+  endif()
+  ExternalProject_Add(liburing_ext
+    DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+    GIT_REPOSITORY http://git.kernel.dk/liburing
+    GIT_TAG "556960942eaa69fd53544932f00db3fa9f196e00"
+    SOURCE_DIR ${CMAKE_BINARY_DIR}/src/liburing
+    CONFIGURE_COMMAND <SOURCE_DIR>/configure
+    BUILD_COMMAND env CC=${CMAKE_C_COMPILER} ${make_cmd} -C src -s
+    BUILD_IN_SOURCE 1
+    INSTALL_COMMAND "")
+  unset(make_cmd)
+  add_library(liburing STATIC IMPORTED GLOBAL)
+  add_dependencies(liburing liburing_ext)
+  set_target_properties(liburing PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_BINARY_DIR}/src/liburing/src/include"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "C"
```

Comment:
in addition to `IMPORTED_LOCATION`, i also attached more properties to this library, so we don't need to set them manually, like
```cmake
target_include_directories(os SYSTEM PRIVATE "${CMAKE_BINARY_DIR}/src/liburing/src/")
```

Reply:
I removed INTERFACE_INCLUDE_DIRECTORIES, this line causes a cmake error:

```
CMake Error in src/os/CMakeLists.txt:
  Imported target "liburing" includes non-existent path

    "/root/rpen/ceph/build/src/liburing/src/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.

```


And also this one "target_include_directories" is required, otherwise build fails.